tests/integration/Application/Validator/Constraints/ValidAttributeValidatorTest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/**
+ * Copyright 2022 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace integration\Application\Validator\Constraints;
+
+use Surfnet\ServiceProviderDashboard\Application\Service\AttributeService;
+use Surfnet\ServiceProviderDashboard\Application\Validator\Constraints\ValidAttributeValidator;
+use Surfnet\ServiceProviderDashboard\Domain\ValueObject\Attribute;
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Repository\AttributeRepository;
+use Surfnet\ServiceProviderDashboard\Application\Validator\Constraints\ValidAttribute;
+use Surfnet\ServiceProviderDashboard\Tests\Integration\Application\Validator\Constraints\mock;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+class ValidAttributeValidatorTest extends ConstraintValidatorTestCase
+{
+    protected function createValidator(): ValidAttributeValidator
+    {
+        $attributeRepository = new AttributeRepository(__DIR__ . '/../fixture/attributes.json');
+        $attributeService = new AttributeService($attributeRepository, 'en');
+        return new ValidAttributeValidator($attributeService);
+    }
+
+    public function test_success_no_attributes()
+    {
+        $constraint = new ValidAttribute();
+        $this->validator->validate([], $constraint);
+
+        $this->assertNoViolation();
+    }
```

Comment:
May I suggest you add another happy flow test. Where you verify it can test multiple attributes? Something along these lines:

```
    public function test_success_multiple_requested_attributes()
    {
        $constraint = new ValidAttribute();
        $this->validator->validate($this->buildMultipleAttributes(
            ['emailAddressAttribute', 'eduPersonTargetedIDAttribute', 'surNameAttribute',]
        ), $constraint);
        $this->assertNoViolation();
    }


    private function buildMultipleAttributes(array $names): array {
        $attributes = [];
        foreach ($names as $name) {
            $requested = true;
            $motivation = 'Motivation';
            $attributes[$name] = new Attribute();
            $attributes[$name]
                ->setRequested($requested)
                ->setMotivation($motivation);
        }
        return $attributes;
    }
```